common/src/main/java/software/amazon/kms/common/AbstractKmsApiHelper.java

```diff
@@ -59,6 +63,12 @@ protected <T> T wrapKmsExceptions(final String operation, final Supplier<T> serv
             throw new CfnNotFoundException(addMessageIfNull(operation, e));
         } catch (final KmsException e) {
             if (ACCESS_DENIED_ERROR_CODE.equals(e.awsErrorDetails().errorCode())) {
+                 // If this is a tagging related Access Denied we need to throw CfnUnauthorizedTaggingOperationException
+                 if (e.getMessage().contains(KMS_TAG_RESOURCE_PERMISSION) ||
+                         e.getMessage().contains(KMS_UNTAG_RESOURCE_PERMISSION)) {
```

Comment:
Should we also add ListResourceTag Permissions here? Without ListResourceTags permission, I don't think we will be able to update tags. I got this when I create a key with tags. Removed ListResourceTags permissions. Tried to update the tags in the templated. The update failed with "Resource handler returned message: "Access denied for operation 'ListResourceTags'." (RequestToken: ea86e3c7-4653-b497-a54f-454202f6fba7, HandlerErrorCode: AccessDenied)"